lib/puppet_x/puppetlabs/scheduled_task/trigger.rb

```diff
@@ -465,15 +467,27 @@ class Type
       TASK_TRIGGER_SESSION_STATE_CHANGE  = 11
     end
 
-    TYPE_MANIFEST_MAP = {
-      Type::TASK_TRIGGER_DAILY => 'daily',
-      Type::TASK_TRIGGER_WEEKLY => 'weekly',
+    TIME_BASED_TRIGGER_MAP = {
+      Type::TASK_TRIGGER_DAILY      => 'daily',
+      Type::TASK_TRIGGER_WEEKLY     => 'weekly',
       # NOTE: monthly uses context to determine MONTHLY or MONTHLYDOW
-      Type::TASK_TRIGGER_MONTHLY => 'monthly',
+      Type::TASK_TRIGGER_MONTHLY    => 'monthly',
       Type::TASK_TRIGGER_MONTHLYDOW => 'monthly',
-      Type::TASK_TRIGGER_TIME => 'once',
+      Type::TASK_TRIGGER_TIME       => 'once',
+    }.freeze
+
+    EVENT_BASED_TRIGGER_MAP = {
+      Type::TASK_TRIGGER_BOOT                 => 'boot',
+      # The triggers below are not yet supported.
+      # Type::TASK_TRIGGER_EVENT                => 'event',
+      # Type::TASK_TRIGGER_IDLE                 => 'idle',
+      # Type::TASK_TRIGGER_REGISTRATION         => 'task_registered',
+      # Type::TASK_TRIGGER_LOGON                => 'logon',
+      # Type::TASK_TRIGGER_SESSION_STATE_CHANGE => 'session_state_change',
```

Comment:
@Iristyle These triggers are commented out so we don't forget them, but if they are not commented there is indeed a bug where a task that has an unsupported trigger type manually applied to it, a puppet run will delete that trigger.

Reply:
And specs didn't catch that problem, right?

That's a little concerning because I know that there are some specs around that at https://github.com/puppetlabs/puppetlabs-scheduled_task/blob/master/spec/unit/puppet/provider/scheduled_task/win32_taskscheduler_spec.rb#L307-L391

We should probably double check that. They may be too mocked to show the real behavior.

Reply:
Yeah I'm not sure just returning nil for an index as in line 317 is sufficient to catch a trigger that exists but is of an unknown type, given the kind of change that I had made.